src/react-native/11-navigation/navigation.md

```diff
@@ -2,23 +2,120 @@
+
+The React Navigation library used in this section supports deep linking and has a [guide](https://reactnavigation.org/docs/deep-linking/) on how to set up your application to be notified of incoming links.
+
+### Stacks
+
+In a web application, routing is typically done through anchor tags. Whenever a user clicks on a link, the URL is pushed to the browser history stack. When the user presses the back button, the browser pops the last URL and navigates to the previously visited page.
+
+*Note: A stack is a data structure that follows the Last In, First Out (LIFO) principle.*
+
+React Native does not have a built-in global history stack similar to a web browser. Luckily, there is a library called React Navigation that provides a way to manage navigation in a stack-like manner.
+
+### Navigation Lifecycle
+
+Now that we have an understanding of how navigation works within a stack, let's take a look at the implications of navigation on the component lifecycle.
+
+If we have a stack navigator with two screens, `ScreenA` and `ScreenB`, and navigate to `ScreenA`, then `ScreenA` will mount the component and render. If we then navigate to `ScreenB`, `ScreenA` will be unmounted and `ScreenB` will mount and render. What happens when we navigate back to `ScreenA`? `ScreenB` will unmount the component, but `ScreenA` will *not* remount. Because it stayed in the stack, it remained mounted the entier time.
+
+### Bottom Tab Navigation
+
+Tab navigation pattern is one of the most common navigation patterns used in both Android and iOS applications. It allows users to switch between different screens by tapping on tabs that are typically located at the bottom of the screen.
+
+Often tabs in a mobile application are made up of more than just one screen. Tab navigation is often used in conjunction with stack navigation to allow for more complex navigation patterns. We will cover how to use React Navigation's stack navigation in the next section.
+
+First, let's setup our application to use tab navigation from React Navigation. To do this, we will install the `@react-navigation/bottom-tabs` package:
+
+```bash
+npm install @react-navigation/bottom-tabs
+```
+
+Here is an example of using tab-based navigation in React Native with React Navigation:
+
+```tsx
+import * as React from 'react';
+import { Text, View } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+
+function HomeScreen() {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Home!</Text>
+    </View>
+  );
+}
+
+function SettingsScreen() {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+      <Text>Settings!</Text>
```

Comment:
Really small detail but I think it’s worthwhile: as much as possible, we want our sample code to have the code they need but not overlap with the actual details in PMO, because otherwise folks might* confuse the sample code in the explanation with what they need to implement in the exercise, or think this is all they need.

So it’s a small detail, but since PMO does have a Settings screen, I would change this to something else.

*I say “might” because I was that person.

Reply:
good call out, just gonna do a very generic screenA/screenB so that it's guaranteed to be no overlap/sample code, and IMO a bit easier to talk through during slides